File: src/nl/mpcjanssen/simpletask/sort/PriorityComparator.java
package nl.mpcjanssen.simpletask.sort;

import nl.mpcjanssen.simpletask.task.Priority;
import nl.mpcjanssen.simpletask.task.Task;

public class PriorityComparator extends ReversableComparator {
    public PriorityComparator(boolean reverse) {
        super(reverse);
    }

    @Override
    public int unreversedCompare(Task a, Task b) {
        Priority prioA = a.getPriority();
        Priority prioB = b.getPriority();

        if (prioA.getCode().equals(prioB.getCode())) {
            return 0;
        } else if (prioA.inFileFormat().equals("")) {
            return 1;
        } else if (prioB.inFileFormat().equals("")) {
            return -1;
        } else {
            return prioA.compareTo(prioB);
        }
    }
}

File: src/main/java/nl/mpcjanssen/simpletask/sort/PriorityComparator.kt
package nl.mpcjanssen.simpletask.sort


import nl.mpcjanssen.simpletask.task.Task
import java.util.*

class PriorityComparator : Comparator<Task> {

    override fun compare(a: Task?, b: Task?): Int {
        if (a === b) {
            return 0
        } else if (a == null) {
            return -1
        } else if (b == null) {
            return 1
        }
        val prioA = a.priority
        val prioB = b.priority

        if (prioA.code == prioB.code) {
            return 0
        } else if (prioA.inFileFormat() == "") {
            return 1
        } else if (prioB.inFileFormat() == "") {
            return -1
        } else {
            return prioA.compareTo(prioB)
        }
    }
}
